### PR TITLE
fix invalidations in REPLCompletions from Static.jl

### PR DESCRIPTION
**Pull request by ranocha**
This should hopefully fix quite some invalidations coming from Static.jl.

<details>

<summary>
Here is the code:

</summary>

```julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.0 (2022-08-17)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.8) pkg> activate --temp

(jl_PDrBpd) pkg> add Static
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `/tmp/jl_PDrBpd/Project.toml`
  [aedffcd0] + Static v0.7.6
    Updating `/tmp/jl_PDrBpd/Manifest.toml`
  [615f187c] + IfElse v0.1.1
  [aedffcd0] + Static v0.7.6

julia> using SnoopCompileCore

julia> invalidations = @snoopr using Static

julia> using SnoopCompile

julia> trees = invalidation_trees(invalidations)
7-element Vector{SnoopCompile.MethodInvalidations}:
...
 inserting !(::False) in Static at ~/.julia/packages/Static/sVI3g/src/Static.jl:427 invalidated:
...
                 150: signature Tuple{typeof(!), Any} triggered MethodInstance for REPL.REPLCompletions.get_value(::Expr, ::Module) (23 children)
```

</details>

As far as I can tell, `get_value` will always return a `Bool` as second argument.

### CODE REVIEW

**Comment by Tokazama** [comment]
> As far as I can tell, get_value will always return a Bool as second argument.

Should we just do this then?

```julia
function get_value(sym::Symbol, fn)::Tuple{Any,Bool}
    isdefined(fn, sym) ? (getfield(fn, sym), true) : (nothing, false)
end
get_value(sym::QuoteNode, fn)::Tuple{Any,Bool} = (sym.value, true)
get_value(sym::GlobalRef, fn)::Tuple{Any,Bool} = get_value(sym.name, sym.mod)
get_value(sym, fn)::Tuple{Any,Bool} = (sym, true)
```

**Comment by ranocha** [comment]
I don't know. Does it fix the invalidation? Or do we still need the type assertion?

**Comment by Tokazama** [comment]
> I don't know. Does it fix the invalidation? Or do we still need the type assertion?

I'm pretty sure that works if every method definition available asserts that, but I don't have enough memory available right now to explicitly test this.

**Review by aviatesk — APPROVED** [review]
This type assertion seems reasonable, as the type of `ex` there is only known at runtime but there are 5 possible matching methods for the `get_value`. As an alternative is to define `get_value(args...) = _get_value(args...)::Tuple{Any,Bool}`.

File: stdlib/REPL/src/REPLCompletions.jl [hunk 1]
```diff
@@ -409,9 +409,9 @@ function get_value(sym::Expr, fn)
     end
     sym.head !== :. && return (nothing, false)
     for ex in sym.args
-        ex, found = get_value(ex, fn)
+        ex, found::Bool = get_value(ex, fn)
```
Comment (aviatesk) [inline stdlib/REPL/src/REPLCompletions.jl:412]:
```suggestion
        exfound = get_value(ex, fn)
        ex = exfound[1]; found = exfound[2]::Bool
```

File: stdlib/REPL/src/REPLCompletions.jl [hunk 2]
```diff
@@ -409,9 +409,9 @@ function get_value(sym::Expr, fn)
...
         !found && return (nothing, false)
-        fn, found = get_value(ex, fn)
+        fn, found::Bool = get_value(ex, fn)
```
Comment (aviatesk) [inline stdlib/REPL/src/REPLCompletions.jl:414]:
```suggestion
        exfound = get_value(ex, fn)
        ex = exfound[1]; found = exfound[2]::Bool
```

**Comment by timholy** [comment]
A more elegant fix might be
```julia
ex, found = get_value(ex, fn)::Tuple{Any, Bool}
```

**Comment by ranocha** [comment]
> A more elegant fix might be
> 
> ```julia
> ex, found = get_value(ex, fn)::Tuple{Any, Bool}
> ```

Thanks! I changed the code according to your suggestion.

**Pull request merged**